parser.go

```diff
@@ -90,25 +94,52 @@ type TemplateFieldSpecifier struct {
 	Length       uint16
 }
 
+// An option can be passed to New()
+type Option func(*Session)
+
+// WithTemplateIDAliasing enables or disables template id aliasing. The default is disabled.
+func WithTemplateIDAliasing(v bool) Option {
+	return func(s *Session) {
+		s.withTemplateIDAliasing = v
+	}
+}
+
 // The Session is the context for IPFIX messages.
 type Session struct {
 	buffers *sync.Pool
 
-	mut       sync.RWMutex
-	templates [][]TemplateFieldSpecifier
-	minRecord []uint16
+	mut sync.RWMutex
```

Comment:
Style: the rule of thumb is that the mutex is a "hat": it sits at the top of the group of things it protects, with any blank line afterwards then meaning that the things following do not belong to the mutex.